packages/data-service-generator/tests/e2e/data-service-generator.e2e.spec.ts

```diff
@@ -193,8 +192,54 @@ describe("Data Service Generator", () => {
             logger.error(err.message, err);
           });
 
+        // Wait for the server to be ready and for database migration before running tests.
+        // TODO replace with a better docker-compose solution that waits for the migraton to be completd before starting the server
+        logger.info("Waiting for db migration to be ...");
+        let migrationCompleted = false;
+        let startTime = Date.now();
+
+        do {
+          logger.info("...");
+          const containers = await compose.ps({
+            ...dockerComposeOptions,
+            commandOptions: ["--all"],
+          });
+          const migrateContainer = containers.data.services.find((s) =>
+            s.name.endsWith("migrate-1")
+          );
+          if (migrateContainer.state.indexOf("Exited (0)") !== -1) {
+            migrationCompleted = true;
+            logger.info("migration completed!");
+            break;
+          }
+          await sleep(1000);
+        } while (
+          !migrationCompleted ||
+          startTime + SERVER_START_TIMEOUT < Date.now()
```

Comment:
Do you want to use the same timeout (SERVER_START_TIMEOUT) for both the migration and the server start?

Reply:
We can make them different but I don't seen any reasons for it at the moment